exec/java-exec/src/test/java/org/apache/drill/exec/physical/impl/svremover/TestSVRemover.java

```diff
@@ -34,4 +38,33 @@ public void testSVRWithNoFilter() throws Exception {
     int numOutputRecords = testPhysical(getFile("remover/sv_with_no_filter.json"));
     assertEquals(100, numOutputRecords);
   }
+
+  /**
+   * Test the generic version of the selection vector remover copier
+   * class. The code uses the traditional generated version by default.
+   * This test sets the option to use the generic version, then runs
+   * a query that exercises that version.
+   * <p>
+   * Note that the tests here exercise only the SV2 version of the
+   * selection remover; no tests exist for the SV4 version.
+   */
+
+  // TODO: Add an SV4 test once the improved mock data generator
+  // is available.
+
+  @Test
+  public void testGenericCopier() throws Exception {
+    // TODO: replace this with new setup once revised test framework
+    // is available.
+    Properties config = new Properties( );
+    config.put(ExecConstants.SYS_STORE_PROVIDER_LOCAL_ENABLE_WRITE, "false");
+    config.put(ExecConstants.HTTP_ENABLE, "false");
+    config.put(ExecConstants.REMOVER_ENABLE_GENERIC_COPIER, "true");
+    updateTestCluster(1, DrillConfig.create(config));
+
+    int numOutputRecords = testPhysical(getFile("remover/test1.json"));
+    assertEquals(50, numOutputRecords);
+    numOutputRecords = testPhysical(getFile("remover/sv_with_no_filter.json"));
+    assertEquals(100, numOutputRecords);
+  }
 }
```

Comment:
It would be good to have the tests that cover all the vector types.  But,  since it is off by default and you are exercising the code through other unit tests,  this is fine. 
+1